csFastFloat/Utils/Utils.cs

```diff
@@ -25,22 +25,28 @@ public static class Utils
     // Next function can be micro-optimized, but compilers are entirely
     // able to optimize it well.
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal static bool is_integer(char c) => (char)(c - '0') <= '9' - '0';
+    internal static bool is_integer(char c, out uint cMinus0)
```

Comment:
The re-use of `cMinus0' produces better code.

Before (a loop):
```asm
M03_L25:
       cmp       rbx,13
       jle       near ptr M03_L33
       mov       byte ptr [rsp+12],1
       xor       r11d,r11d
       mov       rdx,r10
       jmp       short M03_L27
M03_L26:
       lea       r11,[r11+r11*4]
       add       eax,0FFFFFFD0
       lea       r11,[rax+r11*2]
       add       rdx,2
       mov       rax,0B6B3A7640000
       cmp       r11,rax
       jae       short M03_L28
M03_L27:
       cmp       rdx,r8
       je        short M03_L28
       movzx     eax,word ptr [rdx]
       mov       r9d,eax
       add       r9d,0FFFFFFD0
       cmp       r9d,9
       jbe       short M03_L26
```

With this change:
```asm
M03_L22:
       cmp       rdi,13
       jle       near ptr M03_L29
       mov       byte ptr [rsp+12],1
       xor       r11d,r11d
       mov       rdx,r10
       jmp       short M03_L24
M03_L23:
       mov       rax,0B6B3A7640000
       cmp       r11,rax
       jae       short M03_L25
M03_L24:
       cmp       rdx,r8
       je        short M03_L25
       movzx     eax,word ptr [rdx]
       add       eax,0FFFFFFD0
       cmp       eax,9
       ja        short M03_L25
       lea       r11,[r11+r11*4]
       mov       eax,eax
       lea       r11,[rax+r11*2]
       add       rdx,2
       jmp       short M03_L23
```
So one `add eax,0FFFFFFD0` is eliminated and code layed out a bit different.

Need to re-run the benchmarks later today.

Reply:
@gfoidl : just a side question please: May I ask how you've setuped vTune to display source code and JIT side by side.  I've tried without any luck. It always complains about my .pdb(s).   Maybe it's the version I have 2021.1....
Integration within Visual Studio 2019 is also a problem and Intel team are working on it (I wrote to Intel community support on that).

Reply:
I have VTune 2020 Update 3 -- didn't know that there's a newer version.

The key points are
* set the application directory to the build-output dir, as there are the pdb(s) too
* set binaries and sources (marked with the arrow) in the "configure analysis" screen

![grafik](https://user-images.githubusercontent.com/5755834/109178067-c3d97c80-7788-11eb-90fa-e8634ad11ae2.png)

![grafik](https://user-images.githubusercontent.com/5755834/109178201-e7042c00-7788-11eb-9570-d0e675c0289e.png)

![grafik](https://user-images.githubusercontent.com/5755834/109178237-eec3d080-7788-11eb-8709-e10bc40cfb59.png)

(Please apologize that I didn't use your name in the path...)

In the collection log it display some warnings about pdbs too, but so far this never was a problem.
Then in Bottom-up screen double-click on a function (or choose "View Source" from the context menu).

![grafik](https://user-images.githubusercontent.com/5755834/109178603-51b56780-7789-11eb-9008-d019395aaf01.png)

Hitting the "Assembly"-button shows it side by side:

![grafik](https://user-images.githubusercontent.com/5755834/109178883-9e993e00-7789-11eb-84dd-c49f96e6f9ed.png)

If you ran into troubles, don't hesitate to ask.

> Integration within Visual Studio 2019 is also a problem

I tried this some time ago, but since then I use VTune as standalone app. 
BTW: the profiler in Visual Studio is really good for finding hot-methods -- especially in larger applications.

Reply:
i'll give it another try.  Maybe I haven't specified enough source directories....